File: analysis/symbol-light-classes/testData/lightClassByPsi/jvmExposeBoxed/internalDeclarations.java
@kotlin.OptIn(markerClass = {kotlin.ExperimentalStdlibApi.class})
public final class InternalNames /* InternalNames*/ {
  @kotlin.jvm.JvmExposeBoxed()
  @org.jetbrains.annotations.NotNull()
  public final @org.jetbrains.annotations.NotNull() ValueInt jvmNamed(@org.jetbrains.annotations.NotNull() ValueInt);//  jvmNamed(@org.jetbrains.annotations.NotNull() ValueInt)

  @kotlin.jvm.JvmExposeBoxed(jvmName = "explicitBoxedName")
  @org.jetbrains.annotations.NotNull()
  public final @org.jetbrains.annotations.NotNull() ValueInt explicitBoxedName(@org.jetbrains.annotations.NotNull() ValueInt);//  explicitBoxedName(@org.jetbrains.annotations.NotNull() ValueInt)

  @kotlin.jvm.JvmName(name = "ignoredJvmName")
  public final int ignoredJvmName(int);//  ignoredJvmName(int)

  @kotlin.jvm.JvmName(name = "jvmNamed")
  public final int jvmNamed(int);//  jvmNamed(int)

  public  InternalNames();//  .ctor()
}

@kotlin.OptIn(markerClass = {kotlin.ExperimentalStdlibApi.class})
@kotlin.jvm.JvmInline()
public final class ValueInt /* ValueInt*/ {
  private final int i;

  @kotlin.PublishedApi()
  @kotlin.jvm.JvmExposeBoxed()
  public final void publishedApiFun();//  publishedApiFun()

  @kotlin.jvm.JvmExposeBoxed()
  public  ValueInt(int);//  .ctor(int)

  @kotlin.jvm.JvmExposeBoxed()
  public final int getInternalProperty$light_idea_test_case();//  getInternalProperty$light_idea_test_case()

  @kotlin.jvm.JvmExposeBoxed()
  public final int getPublishedApiProperty$light_idea_test_case();//  getPublishedApiProperty$light_idea_test_case()

  @kotlin.jvm.JvmExposeBoxed()
  public final void internalFun$light_idea_test_case();//  internalFun$light_idea_test_case()

  @org.jetbrains.annotations.NotNull()
  public @org.jetbrains.annotations.NotNull() java.lang.String toString();//  toString()

  public boolean equals(@org.jetbrains.annotations.Nullable() @org.jetbrains.annotations.Nullable() java.lang.Object);//  equals(@org.jetbrains.annotations.Nullable() java.lang.Object)

  public final int getI();//  getI()

  public int hashCode();//  hashCode()
}

File: analysis/symbol-light-classes/testData/lightClassByPsi/jvmExposeBoxed/internalDeclarations.kt
// WITH_STDLIB
// LIBRARY_PLATFORMS: JVM

@OptIn(ExperimentalStdlibApi::class)
@JvmInline
@JvmExposeBoxed
value class ValueInt(val i: Int) {
    internal fun internalFun() {}

    internal val internalProperty: Int
        get() = i

    @PublishedApi
    internal fun publishedApiFun() {}

    @PublishedApi
    internal val publishedApiProperty: Int
        get() = i
}

@OptIn(ExperimentalStdlibApi::class)
class InternalNames {
    @JvmExposeBoxed("explicitBoxedName")
    @JvmName("ignoredJvmName")
    internal fun explicitlyNamedFun(value: ValueInt): ValueInt = value

    @JvmExposeBoxed
    @JvmName("jvmNamed")
    internal fun jvmNamedFun(value: ValueInt): ValueInt = value
}

@JvmInline
internal value class InternalValue(val value: Long)

@OptIn(ExperimentalStdlibApi::class)
internal class InternalConstructor @JvmExposeBoxed constructor(
    value: InternalValue,
    callback: () -> Unit = {},
)

// LIGHT_ELEMENTS_NO_DECLARATION: InternalConstructor.class[_init_$lambda$0], InternalNames.class[explicitBoxedName], InternalValue.class[constructor-impl;equals-impl;equals-impl0;hashCode-impl;toString-impl], ValueInt.class[constructor-impl;equals-impl;equals-impl0;getInternalProperty-impl$main;getPublishedApiProperty-impl;hashCode-impl;internalFun-impl$main;publishedApiFun-impl;toString-impl]
